prov/usnic/src/usnic_direct/libnl_utils_common.c

```diff
@@ -98,15 +99,28 @@ static int usnic_is_nlreply_err(struct nlmsghdr *nlm_hdr)
 static int usnic_nl_send_query(struct usnic_nl_sk *unlsk, struct nl_msg *msg,
 				int protocol, int flag)
 {
+	int ret;
 	struct nlmsghdr *nlhdr;
 
 	nlhdr = nlmsg_hdr(msg);
-	nlhdr->nlmsg_pid = nl_socket_get_local_port(unlsk->nlh);
-	nlhdr->nlmsg_seq = ++unlsk->seq;
-	nlmsg_set_proto(msg, protocol);
-	nlhdr->nlmsg_flags = flag;
+	while (1) {
+		nlhdr->nlmsg_pid = nl_socket_get_local_port(unlsk->nlh);
+		nlhdr->nlmsg_seq = ++unlsk->seq;
+		nlmsg_set_proto(msg, protocol);
+		nlhdr->nlmsg_flags = flag;
+
+		/* If nl_send() fails with errno==ECONNREFUSED, it
+		 * just means that it temporarily ran out of
+		 * resources.  Delay a little and try again. */
+		ret = nl_send(unlsk->nlh, msg);
+		if (ret == -1 && errno == ECONNREFUSED) {
```

Comment:
Are you sure that errno gets set by `nl_send`? They don't mention it anywhere in their docs.